source/common/config/grpc_mux_impl.cc

```diff
@@ -222,10 +224,17 @@ void GrpcMuxImpl::onStreamEstablished() {
 }
 
 void GrpcMuxImpl::onEstablishmentFailure() {
-  for (const auto& api_state : api_state_) {
-    for (auto watch : api_state.second.watches_) {
-      watch->callbacks_.onConfigUpdateFailed(
-          Envoy::Config::ConfigUpdateFailureReason::ConnectionFailure, nullptr);
+  // Try to connect all of subscribed clusters which are included on grpc_service. If all of
+  // attempts were failed, It may cause EnvoyException and notify all watches that connection
+  // can't be established.
+  try {
+    callbacks_.updateCluster();
```

Comment:
How does this interact with retry timers? Also, onRemoteClose() seems to call into this method, but we might legitimately want to retry on the existing gRPC service.